executor-service/src/main/java/executor/service/service/scenario/ScenarioHttpService.java

```diff
@@ -27,6 +29,28 @@ public void saveScenarios(List<Scenario> scenarios) {
         scenarioQueue.addAll(scenarios);
     }
 
+    @Override
+    public ResponseEntity<Scenario> getScenarioByUsername(
+            final String username, final String scenarioName) {
+        Scenario scenario = getScenariosByUsername(username)
+                    .stream()
+                    .filter(s -> s.getUsername().equals(username))
```

Comment:
scenarios are filtered by username for the second time in this method. Maybe you wanted this: 
```suggestion
                    .filter(s -> s.getName().equals(scenarioName))
```